src/libstd/io/util.rs

```diff
@@ -45,7 +45,8 @@ use io::{self, Read, Write, ErrorKind, BufRead};
 /// # }
 /// ```
 #[stable(feature = "rust1", since = "1.0.0")]
-pub fn copy<R: Read, W: Write>(reader: &mut R, writer: &mut W) -> io::Result<u64> {
+pub fn copy<R: ?Sized + Read, W: ?Sized + Write>(reader: &mut R, writer: &mut W)
+    -> io::Result<u64> {
```

Comment:
I would personally vote for the ideal style of:

```
pub fn copy<R, W>(reader: &mut R, writer: &mut W) -> io::Result<u64>
    where R: Read + ?Sized, W: Write + ?Sized 
{ 
    /* ... */
}
```

Unfortunately `?Sized` doesn't work in a where clause right now so that's not possible. Overall though I'd go with option (C). I do agree though that we're not consistent throughout the codebase on what style of where clauses we have.